### PR TITLE
Libingester: Add source/page flipping and image previewing

### PR DESCRIPTION
**Pull request by sgnn7**
https://phabricator.endlessm.com/T16343

### CODE REVIEW

File: previewer/bin/hatch-preview [hunk 1]
```diff
@@ -35,3 +35,9 @@ var child = proc.spawn(electron,
 child.on('close', (code) => {
     process.exit(code);
 });
+
+process.on('SIGINT', function() {
```
Comment (wgreenberg) [inline previewer/bin/hatch-preview]: nit: prefer fat arrow function

File: previewer/preview.html [hunk 2]
```diff
@@ -6,7 +6,20 @@
...
   <body>
-    <div id='preview'></div>
+    <nav class='navbar navbar-light bg-faded' id='flip_controls'>
+      <div class='container'>
+        <button type='button' class='btn btn-primary  btn-lg btn-block' onClick='flipPage();'>Flip</button>
+      </div>
+    </nav>
+
+    <span class='border-1 mt-4'>
+
```
Comment (wgreenberg) [inline previewer/preview.html]: nit: this whitespace feels unnecessary

File: previewer/preview_renderer.js [hunk 3]
```diff
@@ -4,20 +4,58 @@ $ = require('jquery')
...
+
+  // Enable controls if we're an html document
+  if (asset.document) {
+    $('#flip_controls').show();
+    $('#source_code_content').text(asset.document).html();
+  } else {
+    $('#flip_controls').hide();
+  }
+
+  $('#preview').html(asset.document || "")
```
Comment (wgreenberg) [inline previewer/preview_renderer.js]: Maybe a message like `No preview available` would be better here

File: previewer/metadata_renderer.js [hunk 4]
```diff
@@ -4,42 +4,57 @@ $ = require('jquery')
 assetMap = main.assetMap
 hatchFolder = main.hatchFolder
 
+const visibleProps = [ "assetID",
```
Comment (wgreenberg) [inline previewer/metadata_renderer.js]: nit: single quotes

File: previewer/preview_renderer.js (hunk 3 above)
Comment (sgnn7) [inline previewer/preview_renderer.js]: @wgreenberg This one is just to make sure we start with a clean canvas and can handle `undefine`s. The code below here will tack on an image to the preview tag too so having any text there wouldn't work.

**Pull request merged**